File: WebApplication1/NgRoot/src/app/app-hospital/models/his.ts
import { Key, BossApiModel } from '@boss/decorator/BossApiModel';
import { BossListField } from '@boss/decorator/list/BossListField';
import { BossListFilter } from '@boss/decorator/list/BossListFilter';
import { BossFormControl } from '@boss/decorator/form/BossFormControl';
import { Validators } from '@angular/forms';
import { BossTextareaComponent } from '@boss/bosscontrol/contorl-components/boss-textarea.component';
import { BossGridDatetimeFieldComponent } from '@boss/bosscontrol/boss-grid/boss-grid-datetime-field.component';
import { DateFormat } from '@boss/DateFormat';
import { BossList } from '@boss/decorator/list/BossList';
import { I18nNamespace } from '@boss/decorator/i18n/i18nNs';
import { BossForm, IBossFormLayout } from '@boss/decorator/form/BossForm';


export function layout(): IBossFormLayout<his> {
  return [
    [x => x.ID, x => x.Name], //排版
  ]
}

@BossApiModel('api/his')
// @I18nNamespace('')
@BossForm(layout())
// tslint:disable-next-line:class-name
export class his {
  @Key()
  public ID: number;

  @BossListField({linkToView: true})
  @BossListFilter()
  @BossFormControl({ validator: Validators.required })
  public Name: string;

  @BossListField({linkToView: true})
  @BossListFilter()
  @BossFormControl({ validator: Validators.required })
  public  Account: string;

  // @BossFormControl({ component: BossTextareaComponent })
  // public  Password: string;

  // @BossFormControl({ component: BossTextareaComponent })
  // public  Status: number;

  // @BossFormControl({ component: BossTextareaComponent })
  // public  Remark: string;

  // @BossListField({ component: BossGridDatetimeFieldComponent, componentData: { format: DateFormat.yMMddHHmmss }, order: 1 })
  // public CreatedTime: Date;

  // @BossFormControl({ component: BossTextareaComponent })
  // public  CreatedBy: number;

  // @BossListField({ component: BossGridDatetimeFieldComponent, componentData: { format: DateFormat.yMMddHHmmss }, order: 1 })
  // public  LastModifiedTime: Date;

  // public  LastModifiedBy: number;
}

File: Bossinfo.Module.Hospital/his.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Bossinfo.Module.Hospital
{
    public class his
    {
        [Key]
        public long ID { get; set; }

        [StringLength(200)]
        public string Name { get; set; }

        [StringLength(10)]
        public string Account { get; set; }

        [StringLength(10)]
        public string Password { get; set; }

        public int Status { get; set; }

        [StringLength(500)]
        public string Remark { get; set; }

        public DateTime? CreatedTime { get; set; }

        public long? CreatedBy { get; set; }

        public DateTime? LastModifiedTime { get; set; }

        public long? LastModifiedBy { get; set; }
    }
}
